src/app/Recordings/ActiveRecordingsTable.tsx

```diff
@@ -930,11 +930,11 @@ export const ActiveRecordingRow: React.FC<ActiveRecordingRowProps> = ({
 
   React.useEffect(() => {
     addSubscription(
-      context.notificationChannel.messages(NotificationCategory.ReportSuccess).subscribe(() => {
-        handleLoadAnalysis();
+      context.notificationChannel.messages(NotificationCategory.ReportSuccess).subscribe((notification) => {
+        if (context.reports.getJobIds().includes(notification.message.jobId)) handleLoadAnalysis();
```

Comment:
It looks like there's a nice opportunity to do some refactoring here. If we give the Reports Service a handle on the Notification Channel also, then the Reports Service can expose an Observable that emits a value whenever an expected long-running Report generation response notification is observed. Then this table, and the automated analysis card, and whatever other UI components need to react can simply subscribe to that Observable instead (rather than each subscribing to the overall notification stream and then checking the Report Service's internal state on their own).

Reply:
I've done some refactoring here bringing it in line with how Targets.service does it (grabbing the NotificationChannel in the constructor and subscribing there) but it's running into an issue at runtime now,

It looks like the notification channel it gets from Services is undefined, trying to access this and subscribe is throwing a TypeError. 

Is there some other initialization step I've missed or something I'm not understanding with Typescript? Why does TargetsService get a valid NotificationChannel but ReportsService doesn't?

Reply:
nevermind, found the isssue, FakeData needed to be updated as well.